### DISCUSSION TITLE
Layout Transition with NextJS Intercepting Routes

### DISCUSSION THREAD

**Discussion by tommyboylab** [motiondivision/motion#2143] [Q&A]
I'm trying to create a shared layout transition similar to the Framer AppStore Example within the NextJS Intercepting Routes example (Nextgram). 

https://github.com/vercel-labs/nextgram
https://codesandbox.io/s/app-store-ui-using-react-and-framer-motion-ecgc2

I've gotten the content to render correctly in both a Modal and a single page, but have not been able to transition the layout from either.

Has anyone got shared Layout transitions working using route Interception? I'll attatch a Gif of my examples.

Working example (without intercepting the route but transitioning page):
![transition 1](https://github.com/framer/motion/assets/44362297/53b8a02d-e537-4f23-b731-bd545b7874eb)

Not working example (with intercepting the route to a modal):
![transition 2](https://github.com/framer/motion/assets/44362297/fe06484b-ada2-427a-8a1a-b86e1d2234f8)

Nextgram example (same as the not working example, but using Nextgram repo as template) but notice transition back only happens once:

![transition 3](https://github.com/framer/motion/assets/44362297/364dc5fd-2dcc-48f7-9d23-1a908b2f0c37)

**Comment by hajirasyafi** [comment]


**Reply by solo-explores** [reply]
can you share it, that repo is no longer available

**Comment by solo-explores** [comment]
did you figure this out?
I too am trying to do something similar, but can't get it right.

**Reply by hajirasyafi** [reply]
can you share your example code on codesandbox?

**Comment by okanji** [comment]
I was wondering the samd, @hajirasyafi can you please re share your solution?

**Comment by lumpinif** [comment]
I am wondering the same, has anyone have the solution?

**Comment by lumpinif** [comment]
@tommyboylab Hey, can you share the examples of the Working example (without intercepting the route but transitioning page) and the Not working example (with intercepting the route to a modal) on codesandbox? I would like to see what I can contribute, if I have the solution, I will immediately post it back with you.

**Comment by artemavrin** [comment]
I just created working example [artemavrin/animation-layout](https://github.com/artemavrin/animation-layout)
and discussion [here ](https://github.com/vercel/next.js/discussions/73047)

**Reply by lumpinif** [reply]
that's cool
